cms/models.py

```diff
@@ -736,3 +740,195 @@ class ResourcePagesSettings(BaseSetting):
+
+    class Meta:
+        abstract = True
+
+    parent_page_types = ["HomePage"]
+    subpage_types = ["SignatoryPage"]
+
+    @classmethod
+    def can_create_at(cls, parent):
+        """
+        You can only create one of these pages under the home page.
+        The parent is limited via the `parent_page_type` list.
+        """
+        return (
+            super().can_create_at(parent)
+            and not parent.get_children().type(cls).exists()
+        )
+
+    def serve(self, request, *args, **kwargs):
+        """
+        For index pages we raise a 404 because these pages do not have a template
+        of their own and we do not expect a page to available at their slug.
+        """
+        raise Http404
+
+
+class SignatoryIndexPage(SignatoryObjectIndexPage):
+    """
+    A placeholder page to group all the signatories under it as well
+    as consequently add /signatories/ to the signatory page urls
+    """
+
+    slug = SIGNATORY_INDEX_SLUG
+
+
+class SignatoryPage(Page):
+    """ CMS page representing a Signatory. """
+
+    promote_panels = []
+    parent_page_types = [SignatoryIndexPage]
+    subpage_types = []
+
+    name = models.CharField(
+        max_length=250, null=False, blank=False, help_text="Name of the signatory."
+    )
+    title_1 = models.CharField(
+        max_length=250,
+        null=True,
+        blank=True,
+        help_text="Specify signatory first title in organization.",
+    )
+    title_2 = models.CharField(
+        max_length=250,
+        null=True,
+        blank=True,
+        help_text="Specify signatory second title in organization.",
+    )
+    organization = models.CharField(
+        max_length=250,
+        null=True,
+        blank=True,
+        help_text="Specify the organization of signatory.",
+    )
+
+    signature_image = models.ForeignKey(
+        Image,
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+        help_text="Signature image size must be at least 150x50 pixels.",
+    )
+
+    class Meta:
+        verbose_name = "Signatory"
+
+    content_panels = [
+        FieldPanel("name"),
+        FieldPanel("title_1"),
+        FieldPanel("title_2"),
+        FieldPanel("organization"),
+        ImageChooserPanel("signature_image"),
+    ]
+
+    def save(self, *args, **kwargs):
+        # auto generate a unique slug so we don't hit a ValidationError
+        if not self.title:
+            self.title = self.__class__._meta.verbose_name.title() + "-" + self.name
+
+        self.slug = slugify("{}-{}".format(self.title, self.id))
+        super().save(*args, **kwargs)
+
+    def serve(self, request, *args, **kwargs):
+        """
+        As the name suggests these pages are going to be children of some other page. They are not
+        designed to be viewed on their own so we raise a 404 if someone tries to access their slug.
+        """
+        raise Http404
+
+
+class CertificatePage(BootcampRunChildPage):
```

Comment:
@asadiqbal08 do you think that we need CertificateIndexPage for certificate routing under HomePage, like we did in xPro?

Reply:
I guess, Yes we need it.